home/HomeApp.tsx

```diff
@@ -19,28 +19,49 @@ import addListenerWithNativeCallback from './utils/addListenerWithNativeCallback
 // Download and cache stack assets, don't block loading on this though
 Asset.loadAsync(StackAssets);
 
+function useSplashScreenWhileLoadingResources(loadResources: () => Promise<void>) {
+  const [isSplashScreenShown, setSplashScreenShown] = React.useState(true);
+  React.useEffect(() => {
```

Comment:
I wrote nearly identical code in my side project (maybe such hook could end up in `expo-splash-screen`? 🤷), but after a second of consideration I changed the hook that prevented autohide from `useEffect` to `useLayoutEffect` — `useEffect`, according to docs is executed

> after the render is committed to the screen

I was afraid this, in some situations, may be too late for the `preventAutoHideAsync` call to execute and thought `useLayoutEffect` should make it more reliable.

I'm not asking you to change it to `useLayoutEffect` if it works, it's just a consideration I went through myself. What do you think? Which hook should be used? Or maybe `SplashScreen.preventAutoHideAsync` should end up in global scope of the app?